src/System.Net.Security/src/System/Net/Unix/CertModule.cs

```diff
@@ -70,14 +71,26 @@ internal override X509Certificate2 GetRemoteCertificate(SafeDeleteContext securi
             try
             {
                 int errorCode = SSPIWrapper.QueryContextRemoteCertificate(GlobalSSPI.SSPISecureChannel, securityContext, out remoteContext);
+
                 if (remoteContext != null && !remoteContext.IsInvalid)
                 {
+                    remoteContext.DangerousAddRef(ref gotReference);
                     result = new X509Certificate2(remoteContext.DangerousGetHandle());
                 }
             }
             finally
             {
-				//TODO: Fetch remoteCertificateStore, if applicable for unix
+                if (gotReference)
+                {
+                    remoteContext.DangerousRelease();
+                }
+
+                if (remoteContext != null && !remoteContext.IsInvalid)
```

Comment:
Even if it's invalid, you probably still want to Dispose it, in case it hasn't yet had finalization suppressed.